test-examples.sh

```
@@ -0,0 +1,350 @@
+    echo "  Building..."
+    local build_output
+    if ! build_output=$(nix build --extra-experimental-features "nix-command flakes" "./$example_dir#default" --no-link 2>&1); then
+        echo -e "${RED}✗ Build failed for $example_name${NC}"
+        echo -e "${RED}Build error output:${NC}"
+        echo "$build_output" | sed 's/^/    /'
+        FAILED_TESTS+=("$example_name: build failed")
+        return 1
+    fi
+    
+    # Get the output path
+    local output_path=$(nix build --extra-experimental-features "nix-command flakes" "./$example_dir#default" --print-out-paths 2>/dev/null)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)